app/solidus_decorators/spree/variant_decorator.rb

```diff
@@ -3,6 +3,8 @@
 
   private
   def mailchimp_sync
-    SolidusMailchimpSync::VariantSynchronizer.new(self).auto_sync
+    unless product.deleted?
```

Comment:
This refers the `acts_as_paranoid` soft delete, there doesn't seem to be an inverse not deleted method or anything, short of doing `!product.deleted?`